### PR TITLE
Testing gobject-sys

### PR DESCRIPTION
**Pull request by EPashkin**
Tried to add gobject-sys, but there it also define `struct GObject`,
may be `struct GObject` generation better skipped?

### CODE REVIEW

**Comment by EPashkin** [comment]
Same problem with `GValue`

**Comment by EPashkin** [comment]
IMHO `use_glib_ffi`, `only_for_gtk` etc.  must be moved to Gir.toml for module as `static_code`

File: src/codegen/sys/lib_.rs [hunk 1]
```diff
@@ -182,6 +183,7 @@ fn generate_interfaces_structs<W: Write>(w: &mut W, interfaces: &[&library::Inte
 fn generate_records<W: Write>(w: &mut W, env: &Env, records: &[&library::Record]) -> Result<()> {
     try!(writeln!(w, ""));
     for record in records {
+        if record.c_type == "GValue" { continue; }
```
Comment (gkoz) [inline src/codegen/sys/lib_.rs]: Is there a reason to have `GObject` and `GValue` defined in `glib-sys` contrary to the gir definitions?

**Comment by gkoz** [comment]
> IMHO use_glib_ffi, only_for_gtk etc. must be moved to Gir.toml for module as static_code

I wouldn't want to put any code in the configs. And the only discrepancy I expect is the predefined types in `glib-sys`, that isn't that big a deal and we could try to minimize the diff.

File: src/codegen/sys/lib_.rs (hunk 1 above)
Comment (EPashkin) [inline src/codegen/sys/lib_.rs]: Its used in glib, interdependency :cry
I'm wrong?

Comment (gkoz) [inline src/codegen/sys/lib_.rs]: I don't think it is. Removing these two definitions doesn't break `glib-sys` for me.

**Comment by gkoz** [comment]
It seems the hardcoded typedefs can actually be limited to `gboolean`, `gpointer` and `gconstpointer`, so an extra module is overkill.

**Comment by gkoz** [comment]
`only_for_gtk` is down to constants so it won't be needed at all.

**Comment by gkoz** [comment]
BTW, I've update the gir-files repo, don't forget to pull.

**Comment by gkoz** [comment]
`Gio` references some `*Private` structs too so the simplest approach will be to not blacklist any structs at all.

**Comment by EPashkin** [comment]
This union in GObject-2.0.gir has no `c:type` so `generate_unions` panics on unwrap.
`<union name="_Value__data__union">`
Write `pub type {} = {}` only for Some?

**Comment by gkoz** [comment]
Meh. Yes please.

**Comment by EPashkin** [comment]
No `GValue` now, maybe from
`WARN:gir::codegen::sys::ffi_type: [Missing glib_name of _Value__data__union, can't match != gpointer]`

**Comment by EPashkin** [comment]
It present but commented

**Comment by EPashkin** [comment]
I merged latest master

**Comment by gkoz** [comment]
I guess I've fooled myself with that `ffi_type` skipping checks deal. If that's the only error, we could just merge despite failing tests.

**Comment by EPashkin** [comment]
After i manual uncomment GValue and add `, GList, GSList` to statics `use glib_ffi::`
We have commented GVaClosureMarshal for va_list and after uncomment last only warning
```
src\lib.rs:59:36: 59:46 warning: literal out of range for i32, #[warn(overflowing_literals)] on by default
src\lib.rs:59         const G_PARAM_DEPRECATED = 2147483648,
```

**Comment by EPashkin** [comment]
Seems ready to merge

**Comment by gkoz** [comment]
Oh right, this PR might need to be rebased onto master and reopened as such..

**Comment by EPashkin** [comment]
After you merge your tests  to master?

**Comment by EPashkin** [comment]
Also I don't like removing `Gir_` prefix from toml.

**Comment by EPashkin** [comment]
Oh, missed, that you merged tests already

**Comment by gkoz** [comment]
> Also I don't like removing Gir_ prefix from toml.

Okay, let's put it back. The capitalization annoys me a bit :)

**Pull request closed without merging**